File: ChakaiP2EAPI/ChakaiP2EFunctionsAPI/Models/Passive.cs
using ChakaiP2EFunctionsAPI.Utilities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChakaiP2EFunctionsAPI.Models
{
    [View]
    public class Passive : BaseModel
    {
        [NoInsert, NoUpdate, Id]
        public int PassiveId { get; set; }
        public string PassiveName { get; set; }
        public string PassiveDescription { get; set; }
        public int SourceTypeId { get; set; }
        [NoInsert, NoUpdate]
        public string SourceTypeName { get; set; }
        [ChildCollection, Xref]
        public List<Trait> Traits { get; set; }
        [ChildCollection, Xref]
        public List<Roll> Rolls { get; set; }
        [ChildCollection, Xref]
        public List<Action> Actions { get; set; }
    }
}

File: ChakaiP2EVue/chakaip2evueapp/src/types/passive.ts
import type { Trait } from './trait';
import type { Action } from './action';

export interface Passive {
    PassiveId: number;
    PassiveName: string;
    PassiveDescription: string;
    PassiveTypeId: number;
    PassiveTypeName: string;

    Traits: Trait[];
    Rolls: { RollId: number }[];
    Actions: Action[];
}
